src/proof_system/verifier.rs

```diff
@@ -89,6 +90,12 @@ impl Verifier {
         public_inputs: &[BlsScalar],
     ) -> Result<(), Error> {
         let mut cloned_transcript = self.preprocessed_transcript.clone();
+
+        // PIs have to be part of the transcript
+        for pi in public_inputs.iter() {
+            cloned_transcript.append_scalar(b"pi", pi);
+        }
```

Comment:
We have a discrepancy between the prover and verifier here.

In the prover, we are using the dense public inputs (that is just the expected public inputs filled with zeroes to match their expected indices).

Here we are using just a not indexed small set of public inputs. Is that intended? If yes, why is that correct?

Reply:
Sounds good, we create a dense PI before coming to this part:

https://github.com/dusk-network/plonk/blob/c6af25f6693e9e06a259a87f321e7dd2298b5c57/src/circuit.rs#L344-L372